# I. Thông tin Dataset

## Nguồn gốc
Dataset được cung cấp bởi **DataCamp**. Dataset được tạo ra thông qua API của `coinmarketcap.com` (vào ngày 06/12/2017).

## Nội dung
Dataset bao gồm các dữ liệu liên qua đến thị trường Cryptocurrencies
___

# II. Khám phá dữ liệu

## Import thư viện cần thiết

Trước hết, import các thư viện cần thiết cho việc khám phá, xử lý và phân tích trở nên dễ dàng hơn.

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Đọc dataset từ file csv
Đọc dữ liệu từ file `coinmarketcap_06122017.csv` được cung cấp

In [18]:
raw_df = pd.read_csv("../data/raw/coinmarketcap_06122017.csv")

Thêm option để các giá trị float dễ đọc hơn


In [19]:
pd.set_option('display.float_format', lambda x: '%8.1f' % x)

In ra 5 dòng đầu tiên của dataset để có thể nắm bắt được khái quát các cột và nội dung của mỗi observation


In [20]:
raw_df.head()

Unnamed: 0  24h_volume_usd  available_supply            id  last_updated  \
0           0    9007640000.0        16723525.0       bitcoin    1512549554   
1           1    1551330000.0        96165368.0      ethereum    1512549553   
2           2    1111350000.0        16840438.0  bitcoin-cash    1512549578   
3           3    2936090000.0      2779530283.0          iota    1512549571   
4           4     231505000.0     38739145009.0        ripple    1512549541   

   market_cap_usd     max_supply          name  percent_change_1h  \
0  213049346738.0     21000000.0       Bitcoin                0.1   
1   43529446198.0            NaN      Ethereum               -0.2   
2   25295852764.0     21000000.0  Bitcoin Cash                1.6   
3   14752245796.0   2779530283.0          IOTA               -2.4   
4    9365343263.0 100000000000.0        Ripple                0.6   

   percent_change_24h  percent_change_7d  price_btc  price_usd  rank symbol  \
0                 7.3               17.4        1.0    12739.5     1    BTC   
1                -3.9               -7.3        0.0      452.7     2    ETH   
2                -5.5               -4.8        0.1     1502.1     3    BCH   
3                83.3              255.8        0.0        5.3     4  MIOTA   
4                -3.7              -14.8        0.0        0.2     5    XRP   

   total_supply  
0    16723525.0  
1    96165368.0  
2    16840438.0  
3  2779530283.0  
4 99993093880.0

Do file csv chứa cả cột Index nên cần bỏ đi để thuận tiện cho việc quan sát


In [21]:
raw_df = raw_df.drop(raw_df.columns[0], axis=1)

raw_df.head()


24h_volume_usd  available_supply            id  last_updated  \
0    9007640000.0        16723525.0       bitcoin    1512549554   
1    1551330000.0        96165368.0      ethereum    1512549553   
2    1111350000.0        16840438.0  bitcoin-cash    1512549578   
3    2936090000.0      2779530283.0          iota    1512549571   
4     231505000.0     38739145009.0        ripple    1512549541   

   market_cap_usd     max_supply          name  percent_change_1h  \
0  213049346738.0     21000000.0       Bitcoin                0.1   
1   43529446198.0            NaN      Ethereum               -0.2   
2   25295852764.0     21000000.0  Bitcoin Cash                1.6   
3   14752245796.0   2779530283.0          IOTA               -2.4   
4    9365343263.0 100000000000.0        Ripple                0.6   

   percent_change_24h  percent_change_7d  price_btc  price_usd  rank symbol  \
0                 7.3               17.4        1.0    12739.5     1    BTC   
1                -3.9               -7.3        0.0      452.7     2    ETH   
2                -5.5               -4.8        0.1     1502.1     3    BCH   
3                83.3              255.8        0.0        5.3     4  MIOTA   
4                -3.7              -14.8        0.0        0.2     5    XRP   

   total_supply  
0    16723525.0  
1    96165368.0  
2    16840438.0  
3  2779530283.0  
4 99993093880.0

## Số dòng và cột

In [40]:
raw_df.shape

(1326, 15)

Dễ dàng thấy rằng dataset bao gồm 1326 dòng và 15 cột

## Ý nghĩa của các dòng

Mỗi dòng tương ứng với một quan sát của một loại coin tại thời điểm cụ thể nào đó. Mỗi dòng cung cấp các thông tin như giá cả, khối lượng giao dịch, bậc...


## Ý nghĩa của các cột

Để hiểu rõ hơn về nội dung của dataset, ta cần phải giải thích ý nghĩa của từng cột trong dataset. 

- `24h_volume_usd`: khối lượng giao dịch trong 24h

- `available_supply`: số lượng coins đã được phát hành và đang tồn tại trong thị trường ngay bây giờ

- `id`: id để phân biệt các coins với nhau

- `last_updated`: thời gian được cập nhật

- `market_cap_usd`: giá trị vốn hoá tính theo USD

- `max_supply`: ước tính về số lượng coins tối đa có thể tạo ra trong vòng đời của nó.

- `name`: tên của coin.

- `percent_change_1h`: mức độ thay đổi về giá của coin trong vòng 1h (%)

- `percent_change_24h`: mức độ thay đổi về giá của coin trong vòng 24h (%)

- `percent_change_7d`: mức độ thay đổi về giá của coin trong vòng 7d (%)

- `price_btc`: giá của coin tính bằng Bitcoin

- `price_usd`: giá của coin tính bằng USD

- `rank`: bậc của coin theo tiêu chí của coinmarketcap

- `symbol`: là ký hiệu của coin

- `total_supply`: ước tính số lượng coins tối đa (bao gồm phần đang lưu thông và chưa được phát hành) ngay bây giờ.

**Lưu ý**
Có 2 cột khá dễ nhầm lẫn là `total_supply` và `available_supply`.

`total_supply`:
- Bao gồm số lượng coin đang trong lưu thông và phần chưa được phát hành
- Thể hiện số lượng coin tối đa có thể cung cấp

`available_supply`:
- Bao gồm số lượng coin đang trong lưu thông (đã được phát hành)
- Là một phép đo tốt hơn để tính giá trị vốn hoá (capitalization) cho coin đó.





## Kiểm tra sự trùng lắp của các dòng


Để kiểm tra giữa các dòng có sự trùng lắp hay không ta thực hiện như sau:

In [24]:
raw_df.duplicated().sum()

0

Output là 0 cho thấy không có sự trùng lắp giữa các dòng.

## Liệu kiểu dữ liệu của các cột đã hợp lý hay chưa?

Trước hết, quan sát kiểu dữ liệu khi đọc lên từ file csv

In [41]:
raw_df.dtypes

24h_volume_usd        float64
available_supply      float64
id                     object
last_updated            int64
market_cap_usd        float64
max_supply            float64
name                   object
percent_change_1h     float64
percent_change_24h    float64
percent_change_7d     float64
price_btc             float64
price_usd             float64
rank                    int64
symbol                 object
total_supply          float64
dtype: object